### ISSUE TITLE
also lint on items after statements in macros

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust-clippy#578]
This is currently disabled, because it would make e.g. the `regex!` macro very annoying. Should probably be turned on after `regex!` has been fixed.

**Comment by Ryan1729** [comment]
The `regex!` macro seems to be no more. It's unclear how many other macros would trigger this lint, now though.

Just because I had to edit github urls to find it for context, [here's a link to the original lint on items after statements PR](https://github.com/rust-lang/rust-clippy/commit/30c7ea857c5103de9d89f70f014b018afd1cff7e)

**Comment by flip1995** [comment]
```rust
#![warn(clippy::items_after_statements)]

macro_rules! m {
    () => {{
        f();
        fn f() { println!("Hi"); }
    }};
}

fn main() {
    m!();
}
```

Does not emit a warning currently: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4cca1d2c022a950c2685345f145d21ac). I don't see a reason, why we shouldn't lint this in local macros.
